pkg/http/api.go

```diff
@@ -698,6 +733,133 @@ func (a *api) getLockStoreWithRequestValidation(reqCtx *fasthttp.RequestCtx) (lo
 	return a.lockStores[storeName], storeName, nil
 }
 
+// Route:   "workflows/{workflowComponent}/{workflowType}/{instanceId}",
+// Workflow Component: Component specified in yaml (temporal, etc..)
+// Workflow Type: Name of the workflow to run (function name)
+// Instance ID: Identifier of the specific run
+func (a *api) onStartWorkflow(reqCtx *fasthttp.RequestCtx) {
+	startReq := wfs.StartRequest{}
+
+	wfType := reqCtx.UserValue(workflowType).(string)
+	if wfType == "" {
+		log.Error("No workflow, or empty workflow was provided.")
+		return
+	}
+
+	component := reqCtx.UserValue(workflowComponent).(string)
+	if component == "" {
+		log.Error("No component, or empty component was provided.")
+		return
+	}
+	if a.workflows[component] == nil {
+		log.Error("The provided component does not exist.")
+		return
+	}
+
+	err := json.Unmarshal(reqCtx.PostBody(), &startReq)
+	if err != nil {
+		log.Errorf("Error while unmarshalling workflow info: %v\n", err)
+	}
+
+	instance := reqCtx.UserValue(instanceID).(string)
+	if instance == "" {
+		log.Error("No instance, or empty instance was provided.")
+		return
+	}
+
+	req := workflows.StartRequest{
+		Options:      startReq.Options,
+		WorkflowName: wfType,
+		Parameters:   startReq.Parameters,
+	}
+	req.WorkflowInfo.InstanceID = instance
+
+	resp, err := a.workflows[component].Start(reqCtx, &req)
+	if err != nil {
+		log.Errorf("Error when starting workflow: %v\n", err)
+		return
+
+	}
+	response, err := json.Marshal(resp)
+	if err != nil {
+		log.Errorf("Error when marshalling output response from workflow start: %v\n", err)
+		return
+	}
+	log.Debug(resp)
+	respond(reqCtx, withJSON(202, response))
+}
+
+func (a *api) onGetWorkflow(reqCtx *fasthttp.RequestCtx) {
+
+	wfName := reqCtx.UserValue(workflowType).(string)
+	if wfName == "" {
+		log.Error("No workflow, or empty workflow was provided.")
+		return
+	}
+
+	instance := reqCtx.UserValue(instanceID).(string)
+	if instance == "" {
+		log.Error("No instance, or empty instance was provided.")
```

Comment:
Same comment about "instance" --> "instance ID". Messages constants, as mentioned earlier, might be useful to ensure that we're being consistent with all our error messages across API implementations.